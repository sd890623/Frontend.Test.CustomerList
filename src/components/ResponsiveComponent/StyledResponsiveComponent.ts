import styled from "styled-components";

export const StyledMobileComponent = styled.div`
  width:100%;
  @media (max-width: 768px) {
    display: block;
  }
  @media (min-width: 769px) {
    display: none;
  }
`;

export const StyledDesktopComponent = styled.div`
  width:100%;
  @media (max-width: 768px) {
    display: none;
  }
  @media (min-width: 769px) {
    display: block;
  }
`;
// @todo better device width/group