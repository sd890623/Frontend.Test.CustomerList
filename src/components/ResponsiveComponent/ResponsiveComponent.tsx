import {FC, ReactNode} from "react";
import {StyledDesktopComponent, StyledMobileComponent} from "./StyledResponsiveComponent";

interface Props {
    mobile: ReactNode
    desktop: ReactNode
}
const ResponsiveComponent: FC<Props> = (props:Props)=>{
    return <>
        <StyledMobileComponent test-id="mobile-view">
            {props.mobile}
        </StyledMobileComponent>
        <StyledDesktopComponent test-id="desktop-view">
            {props.desktop}
        </StyledDesktopComponent>
    </>
}

export default ResponsiveComponent