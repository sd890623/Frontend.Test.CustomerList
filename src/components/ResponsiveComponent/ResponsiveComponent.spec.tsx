import { render } from "../../utils/testUtils";
import ResponsiveComponent from "./ResponsiveComponent";

describe("<ResponsiveComponent />", () => {
    it("should render a <Accordion />", () => {
        const wrapper = render(<ResponsiveComponent mobile={<p>Mobile</p>} desktop={<p>Desktop</p>}/>);
        expect(wrapper.container).toMatchSnapshot();
    })
    it("should toggle the content in or out by clicking button", () => {
        const { queryByText }= render(<ResponsiveComponent mobile={<p>Mobile</p>} desktop={<p>Desktop</p>}/>);
        expect(queryByText('Mobile')).toBeVisible();
        expect(queryByText('Desktop')).toBeVisible();
    })
    // @todo better testing for mobile/desktop media query
});
