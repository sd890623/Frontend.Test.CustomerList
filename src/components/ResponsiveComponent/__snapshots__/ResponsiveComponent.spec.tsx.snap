// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveComponent /> should render a <Accordion /> 1`] = `
<div>
  <div
    class="sc-bdnxRM iCFseo"
  >
    <p>
      Mobile
    </p>
  </div>
  <div
    class="sc-gtsrHT ebaTNJ"
  >
    <p>
      Desktop
    </p>
  </div>
</div>
`;
